requests/exceptions.py

```diff
@@ -33,6 +33,9 @@ class InvalidJSONError(RequestException):
 
 class JSONDecodeError(InvalidJSONError, CompatJSONDecodeError):
     """Couldn't decode the text into json"""
+    def __init__(self, *args, **kwargs):
+        CompatJSONDecodeError.__init__(self, *args)
+        InvalidJSONError.__init__(self, *self.args, **kwargs)
```

Comment:
Can we add comments here explaining why:

1. We're using super old Python 2.5 methods of calling parent `__init__` methods
2. We're calling the `__init__`s in this particular order - presuming the order matters
3. This fixes the issue of `.doc` ending up in the message/default `__repr__`